syntax = "proto3";

service WorkerService {
    rpc ProcessOrder(OrderRequest) returns (OrderResponse);
}

message OrderRequest {
    string order_id = 1;
    string customer_id = 2;
    repeated Item items = 3;
}

message OrderResponse {
    string order_id = 1;
    bool success = 2;
    string message = 3;
}

message Item {
    string product_id = 1;
    int32 quantity = 2;
    float price = 3;
}
